### ISSUE TITLE
"TextEditor is closed/disposed" messages when using setDecorations

### ISSUE THREAD

**Issue by DanTup** [Microsoft/vscode#32408]
I'm using `setDecorations` to render some labels in editors. I've noticed that I randomly see warnings in the dev tools and in the debug console of the main instance of Code (when I'm debugging using the extension host).

I'm calling the code in a timeout so it's very possible the editor is gone away, however I'm trying to guard against this with the following code:

```ts
private update() {
	const labels = this.closingLabels;
	const editor = this.activeEditor;

	if (!labels
		|| editor != vs.window.activeTextEditor
		|| editor.document.isClosed
		|| labels.file != editor.document.fileName)
		return;

	// (snip build some decorations from labels)

	editor.setDecorations(decType, decs);
```

Is there a race here, or is there a better way for me to check whether the editor is valid for setting decorations on?

```text
[Extension Host] Error: TextEditor disposed
    at t.$trySetDecorations (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:28:3140408)
    at t.e.handle (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:28:3100044)
    at s (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:28:1029376)
    at f (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:28:1030031)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
t.logExtensionHostMessage @ extensionHost.ts:335
(anonymous) @ extensionHost.ts:188
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
```

**Comment by alexandrudima** [comment]
I think we should just remove the warning on our side. It is possible that, by the time the message makes it from the extension host to the renderer process, the editor has been closed.

**Comment by alexandrudima** [comment]
This has been coincidentally fixed already via #30578
